codegen/codegen.go

```diff
@@ -1818,17 +1840,20 @@ func (cg *CodeGen) EmitMountOptions(ctx context.Context, scope *parser.Scope, op
 
 // Get a consistent hash for this local (path + options) so we don't transport
 // the same content multiple times when referenced repeatedly.
-func buildLocalID(ctx context.Context, path string, opts ...llb.LocalOption) (string, error) {
-	// First get serialized bytes for this llb.Local state.
-	st := llb.Local("", opts...)
-	_, hashInput, _, err := st.Output().Vertex(ctx).Marshal(ctx, &llb.Constraints{})
+func (cg *CodeGen) LocalID(path string, opts ...llb.LocalOption) (string, error) {
+	opts = append(opts, llb.SessionID(cg.SessionID()))
+	st := llb.Local(path, opts...)
+
+	def, err := st.Marshal(context.Background())
 	if err != nil {
 		return "", err
 	}
 
-	// Next append the path so we have the path + options serialized hash input.
-	hashInput = append(hashInput, []byte(path)...)
-	return string(digest.FromBytes(hashInput)), nil
+	return digest.FromBytes(def.Def[len(def.Def)-1]).String(), nil
```

Comment:
Heh, this is incoherent, in your PR comment you added:
```
The terminal op of the graph def.Def[len(def.Def)-1] is an empty vertex with an input to the last vertex's digest. Since that vertex also has its digests of its inputs and so on, the digest of the terminal op is a merkle hash for the graph.
```
maybe a similar comment should be in code as well?